pkg/process/procutil/process_linux.go

```diff
@@ -244,26 +318,61 @@ func (p *Probe) parseStatusKV(key, value string, sInfo *statusInfo) {
 			sInfo.ctxSwitches.Involuntary = v
 		}
 	case "VmRSS":
-		value := strings.Trim(value, " kB") // remove last "kB"
+		value := strings.Trim(value, " kB") // trim spaces and suffix "kB"
 		v, err := strconv.ParseUint(value, 10, 64)
 		if err == nil {
 			sInfo.memInfo.RSS = v * 1024
 		}
 	case "VmSize":
-		value := strings.Trim(value, " kB") // remove last "kB"
+		value := strings.Trim(value, " kB") // trim spaces and suffix "kB"
 		v, err := strconv.ParseUint(value, 10, 64)
 		if err == nil {
 			sInfo.memInfo.VMS = v * 1024
 		}
 	case "VmSwap":
-		value := strings.Trim(value, " kB") // remove last "kB"
+		value := strings.Trim(value, " kB") // trim spaces and suffix "kB"
 		v, err := strconv.ParseUint(value, 10, 64)
 		if err == nil {
 			sInfo.memInfo.Swap = v * 1024
 		}
 	}
 }
 
+// ensurePathReadable ensures that the current user is able to read the path before opening it.
+// On some systems, attempting to open a file that the user does not have permission is problematic for
+// customer security auditing. What we do here is:
+// 1. If the agent is running as root (real or via sudo), allow the request
+// 2. If the file is a not a symlink and has the other-readable permission bit set, allow the request
+// 3. If the owner of the file/link is the current user or effective user, allow the request.
+func (p *Probe) ensurePathReadable(path string) error {
+	// User is (effectively or actually) root
+	if p.euid == 0 {
+		return nil
+	}
+
+	// TODO (sk): Provide caching on this!
```

Comment:
This is a bit tricky to cache. We don't have a clear view of which processes are gone at which point, and a same pid could be reused by multiple processes. One thought is that if we get a cache hit, also compare the `start time` of current process with the one that's in the cache, if `start_time > cache_time` then we refresh the cache, otherwise reuse.